opensearchapi/api_indices-create.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+
+package opensearchapi
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+)
+
+// IndicesCreateReq represents possible options for the index create request
+type IndicesCreateReq struct {
+	Index  string
+	Body   io.Reader
```

Comment:
Yes, it would be useful but this is not my goal with this PR and would make it even more complex to review. I want to make the basic refactor of the lib to the named issue.  You could open an issue for that request so we can work on that in the future.